### ISSUE TITLE
`no-unused-properties` - add support for unused props in types

### ISSUE THREAD

**Issue by qpwo** [sindresorhus/eslint-plugin-unicorn#1551]
Not sure if this is a bug exactly but it's not a new rule proposal either.

This doesn't give any errors but I wish it did.

```ts
function foo(args: {x: number, y: number}) {
    return args.x * 2 // y is unused
}
```
